src/main/java/org/example/domain/customer/Customer.java

```diff
@@ -1,21 +1,339 @@
+        }
+
+        BigDecimal monthlyRevenue = incomeInvoices.stream()
+                .filter(date -> yearMonth
+                        .equals(
+                                YearMonth.of(
+                                        date.getDate().getYear(),
+                                        date.getDate().getMonth()
+
+                                )))
+                .map(invoice -> invoice.getAmount().amount())
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+
+        return new Money(monthlyRevenue, Currency.PLN);
+    }
+
+    public Money countMonthlyCosts(YearMonth yearMonth) {
+        if (yearMonth.isAfter(YearMonth.now())) {
+            log.error("We cannot calculate the revenue for the specified month [{}]," +
+                    " because now we have [{}] and it is impossible to calculate" +
+                    "revenue for the upcoming months!", yearMonth, YearMonth.now());
+            throw new CountMonthlyCostsException("We cannot calculate the revenue for the upcoming months!");
+        }
+        BigDecimal monthlyCosts = costInvoices.stream()
+                .filter(date -> yearMonth
+                        .equals(
+                                YearMonth.of(
+                                        date.getDate().getYear(),
+                                        date.getDate().getMonth()
+
+                                )))
+                .map(invoice -> invoice.getAmount().amount())
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+
+        return new Money(monthlyCosts, Currency.PLN);
+    }
+
+    public Money countMonthlyIncome(YearMonth yearMonth) {
+        if (yearMonth.isAfter(YearMonth.now())) {
+            log.error("We cannot calculate the revenue for the specified month [{}]," +
+                    " because now we have [{}] and it is impossible to calculate" +
+                    "revenue for the upcoming months!", yearMonth, YearMonth.now());
+            throw new CountMonthlyIncomeException("We cannot calculate the revenue for the upcoming months!");
+        }
+
+        Money monthlyRevenue = countMonthlyRevenue(yearMonth);
+        Money monthlyCosts = countMonthlyCosts(yearMonth);
+        Money tax = countIncomeTax(yearMonth);
+        BigDecimal finalIncome = monthlyRevenue.amount().subtract(monthlyCosts.amount()).subtract(tax.amount());
+        return new Money(finalIncome, Currency.PLN);
+
+    }
+
+    public Money countIncomeTax(YearMonth yearMonth) {
+        Money monthlyRevenue = countMonthlyRevenue(yearMonth);
+        Money monthlyCosts = countMonthlyCosts(yearMonth);
+
+        BigDecimal incomeBeforeTaxIncludingSocialContributions = countIncomeBeforeTaxIncludingSocialContributions(monthlyRevenue, monthlyCosts);
+        TaxPaymentForm customerTaxPaymentForm = getCustomerTaxPaymentForm(this);
+
+        if (customerTaxPaymentForm instanceof FlatTax) {
```

Comment:
switch case bedzie czytelniesze?

Reply:
We discussed on our previous meeting.
Resolved.